k2/python/k2/autograd.py

```diff
@@ -388,6 +515,19 @@ def intersect_dense_pruned(a_fsas: Fsa, b_fsas: DenseFsaVec,
         frame for any given intersection/composition task. This is advisory,
         in that it will try not to exceed that but may not always succeed.
         You can use a very large number if no constraint is needed.
+      seqframe_idx_name:
+        If set (e.g. to 'seqframe'), an attribute in the output will be created
+        that encodes the sequence-index and the frame-index within that sequence;
+        this is equivalent to a row-index into b_fsas.values, or, equivalently,
+        an element in b_fsas.shape.
+        # TODO: actually implement this.   It can be done by dividing the
```

Comment:
How do we handle the arcs that enter the final state? 

Should we set `seqframe_idx` and `frame_idx` of them to -1?

Reply:
No, the indexes will be set the same way as others, no special case.  In the data that we give to the dense FSA, those frames are like normal frames.

Reply:
Please have a look at the following pull-request
https://github.com/danpovey/k2/pull/2